--- qt-everywhere-opensource-src-4.8.7/src/plugins/accessible/widgets/itemviews.cpp	2015-05-07 09:14:48.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.7/src/plugins/accessible/widgets/itemviews.cpp.static_cast	2020-09-01 09:20:21.737615931 -0500
@@ -393,7 +393,7 @@ bool QAccessibleTable2::unselectColumn(i
     QModelIndex index = view()->model()->index(0, column, view()->rootIndex());
     if (!index.isValid() || view()->selectionMode() & QAbstractItemView::NoSelection)
         return false;
-    view()->selectionModel()->select(index, QItemSelectionModel::Columns & QItemSelectionModel::Deselect);
+    view()->selectionModel()->select(index, static_cast<QItemSelectionModel::SelectionFlags>(QItemSelectionModel::Columns & QItemSelectionModel::Deselect));
     return true;
 }
 
