--- /opt/google/gehttpd/conf/gehttpd.conf	2017-03-17 11:46:10.021258844 -0700
+++ /opt/google/gehttpd/conf/gehttpd.conf	2017-03-17 15:16:15.965546556 -0700
@@ -124,8 +124,8 @@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User IA_GEAPACHE_USER
-Group IA_GEGROUP
+User geapacheuser
+Group gegroup
 
 </IfModule>
 </IfModule>
@@ -385,15 +385,6 @@
 #
 DefaultType text/plain
 
-<IfModule headers_module>
-    #
-    # Avoid passing HTTP_PROXY environment to CGI's on this or any proxied
-    # backend servers which have lingering "httpoxy" defects.
-    # 'Proxy' request header is undefined by the IETF, not listed by IANA
-    #
-    RequestHeader unset Proxy early
-</IfModule>
-
 <IfModule mime_module>
     #
     # TypesConfig points to the file containing the list of mappings from
