--- /etc/init.d/geserver	2017-03-10 17:25:53.029868000 -0800
+++ /etc/init.d/geserver	2017-03-17 15:49:31.235219569 -0700
@@ -41,9 +41,9 @@
 # As of 3.1, GEE Fusion and GEE Server supports custom usernames which are
 # specified at install time.
 #"geapacheuser"
-GEAPACHEUSER="IA_GEAPACHE_USER"
+GEAPACHEUSER="geapacheuser"
 #"gepguser"
-GEPGUSER="IA_GEPGUSER"
+GEPGUSER="gepguser"
 
 prog=geserver
 gehttpd_root=/opt/google/gehttpd/bin
