diff -uNr gdal-1.3.0.orig/frmts/jp2kak/jp2kakdataset.cpp gdal-1.3.0/frmts/jp2kak/jp2kakdataset.cpp
--- gdal-1.3.0.orig/frmts/jp2kak/jp2kakdataset.cpp	2005-07-05 15:09:50.000000000 -0700
+++ gdal-1.3.0/frmts/jp2kak/jp2kakdataset.cpp	2005-10-19 11:27:30.000000000 -0700
@@ -1084,7 +1084,7 @@
 /* -------------------------------------------------------------------- */
 /*      Check header                                                    */
 /* -------------------------------------------------------------------- */
-    if( poOpenInfo->fp == NULL )
+    if( poOpenInfo->nHeaderBytes == 0 )
     {
         pszExtension = CPLGetExtension( poOpenInfo->pszFilename );
         if( (EQUALN(poOpenInfo->pszFilename,"http://",7)
@@ -1354,7 +1354,7 @@
 /* -------------------------------------------------------------------- */
 /*      Look for supporting coordinate system information.              */
 /* -------------------------------------------------------------------- */
-        if( poOpenInfo->fp != NULL )
+        if( poOpenInfo->nHeaderBytes > 0 )
         {
             GDALJP2Metadata oJP2Geo;
             FILE *fpLL;
