--- gdal-1.4.2-orig/configure	2008-06-04 10:39:42.000000000 -0700
+++ gdal-1.4.2/configure	2008-06-04 10:40:12.000000000 -0700
@@ -25269,14 +25269,14 @@
 
 else
 
-  echo "$as_me:$LINENO: checking for lt_base.h in $with_mrsid/include/support" >&5
-echo $ECHO_N "checking for lt_base.h in $with_mrsid/include/support... $ECHO_C" >&6
-  if test -r "$with_mrsid/include/support/lt_base.h" ; then
+  echo "$as_me:$LINENO: checking for lt_base.h in $with_mrsid/include" >&5
+echo $ECHO_N "checking for lt_base.h in $with_mrsid/include... $ECHO_C" >&6
+  if test -r "$with_mrsid/include/lt_base.h" ; then
     echo "$as_me:$LINENO: result: found MrSID DSDK version 4.x or newer." >&5
 echo "${ECHO_T}found MrSID DSDK version 4.x or newer." >&6
 
     HAVE_MRSID=yes
-    MRSID_INCLUDE="-I$with_mrsid/include/base -I$with_mrsid/include/metadata -I$with_mrsid/include/mrsid_readers -I$with_mrsid/include/j2k_readers -I$with_mrsid/include/support"
+    MRSID_INCLUDE="-I$with_mrsid/include"
     MRSID_LIBS="-lltidsdk -lpthread"
 
     echo "$as_me:$LINENO: checking for MG3ImageWriter.h in $with_mrsid/include/mrsid_writers" >&5
