--- httpd-2.2.32.orig/docs/conf/extra/httpd-ssl.conf.in	2015-05-27 11:59:59.000000000 -0700
+++ httpd-2.2.32/docs/conf/extra/httpd-ssl.conf.in	2017-06-15 22:14:57.009538141 -0700
@@ -49,7 +49,9 @@
 #   ensure these follow appropriate best practices for this deployment.
 #   httpd 2.2.30, 2.4.13 and later force-disable aNULL, eNULL and EXP ciphers,
 #   while OpenSSL disabled these by default in 0.9.8zf/1.0.0r/1.0.1m/1.0.2a.
-SSLCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
+#SSLCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
+# Override SSL configuration to require more secure ciphers.
+SSLCipherSuite ALL:!aNULL:!ADH:!eNULL:!LOW:!EXP:RC4+RSA:+HIGH:+MEDIUM
 SSLProxyCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
 
 #  By the end of 2016, only TLSv1.2 ciphers should remain in use.
@@ -76,7 +76,9 @@
 #   Disable SSLv3 by default (cf. RFC 7525 3.1.1).  TLSv1 (1.0) should be
 #   disabled as quickly as practical.  By the end of 2016, only the TLSv1.2
 #   protocol or later should remain in use.
-SSLProtocol all -SSLv3
+#SSLProtocol all -SSLv3
+# Limit SSL protocols.
+SSLProtocol -ALL +TLSv1.2 +TLSv1
 SSLProxyProtocol all -SSLv3
 
 #   Pass Phrase Dialog:
@@ -112,8 +116,8 @@
 
 #   General setup for the virtual host
 DocumentRoot "@exp_htdocsdir@"
-ServerName www.example.com:@@SSLPort@@
-ServerAdmin you@example.com
+#ServerName www.example.com:@@SSLPort@@
+#ServerAdmin you@example.com
 ErrorLog "@exp_logfiledir@/error_log"
 TransferLog "@exp_logfiledir@/access_log"
 
